qiskit/circuit/operation.py

```diff
@@ -0,0 +1,62 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Quantum Operation Mixin."""
+
+from abc import ABC
+from qiskit.circuit.exceptions import CircuitError
+
+
+class Operation(ABC):
+    """Quantum Operation Mixin Class.
+    For objects that can be added to a :class:`~qiskit.circuit.QuantumCircuit`.
+    These objects include other :class:`~qiskit.circuit.QuantumCircuit`,
+    :class:`~qiskit.circuit.Gate`, :class:`~qiskit.circuit.Reset`,
+    :class:`~qiskit.circuit.Barrier`, :class:`~qiskit.circuit.Measure`,
+    and operators such as :class:`~qiskit.quantum_info.Clifford`.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        if cls is Operation:
+            raise CircuitError("An Operation mixin should not be instantiated directly.")
+        return super().__new__(cls)
```

Comment:
I'm not super familiar with using `__new__` but should this include the args and kwargs?

Reply:
The `__new__` gets passed the same arguments as `__init__`. These are usually passed to `super().__new__` but doing that here causes an error for some reason.